### PR TITLE
Simplify build NuGet script

### PR DESCRIPTION
**Pull request by aparajit-pratap**
### Purpose
Simplify build NuGet script.

### Declarations

Check these if you believe they are true

- [x] The code base is in a better state after this PR
- [ ] Is documented according to the [standards](https://github.com/DynamoDS/Dynamo/wiki/Coding-Standards)
- [ ] The level of testing this PR includes is appropriate
- [ ] User facing strings, if any, are extracted into `*.resx` files
- [ ] All tests pass using the self-service CI.
- [ ] Snapshot of UI changes, if any.
- [ ] Changes to the API follow [Semantic Versioning](https://github.com/DynamoDS/Dynamo/wiki/Dynamo-Versions), and are documented in the [API Changes](https://github.com/DynamoDS/Dynamo/wiki/API-Changes) document.

### FYIs

@QilongTang

### CODE REVIEW

File: tools/NuGet/BuildPackages.bat [hunk 1]
```diff
@@ -39,27 +39,13 @@ set version=%Major%.%Minor%.%Build%-beta%Revision%
...
-    setlocal EnableDelayedExpansion
-    set line=!line:@VERSION@=%version%!
-    echo !line!>>nuspec\%%~nxf
-    endlocal
-  )
-)
 
 :: Pack .nupkg files based on each .nuspec in the "nuspec" folder
-for %%f in (nuspec\*.nuspec) do (
+for %%f in (%1\*.nuspec) do (
```
Comment (mjkkirschner) [inline tools/NuGet/BuildPackages.bat]: hmm, I assume that not everyone on the team will be a regular .bat file author - can you add a comment for this syntax.

also... what is the main difference in behavior after this PR?

Comment (aparajit-pratap) [reply tools/NuGet/BuildPackages.bat]: There is no difference in behavior. Previously there was complex batch scripting logic to parse the nuspec file, search for the keyword, `@VERSION@` and then replace it with the value of the `%version%` variable obtained in the previous lines, and finally make a copy of the nuspec file with this version number in a separate folder called `nuspec`. This change is to get rid of this logic by simply parameterizing the version in the nuspec file and setting the parameter to the version number obtained in the bat file. This is done using a `-properties` option while using the `nuget pack` command. I guess the only difference is that after this change there is no need for the redundant nuspec folder. The nuspecs from the original folders can be used, namely, `template-nuget` and `template-artifactory`. These are passed as command line arguments (which is what the %1 is) while invoking the bat script.

Comment (QilongTang) [reply tools/NuGet/BuildPackages.bat]: Make me think worth noting this down in https://github.com/DynamoDS/Dynamo/blob/master/tools/NuGet/README.md

**Comment by QilongTang** [comment]
LGTM

**Pull request merged**